qt/scientific_interfaces/Indirect/test/IndirectFitDataTest.h

```diff
@@ -0,0 +1,447 @@
+  }
+
+  template <typename T> bool operator()(const T &lhs, const T &rhs) const {
+    return lhs == rhs;
+  }
+};
+
+} // namespace
+
+class IndirectFitDataTest : public CxxTest::TestSuite {
+public:
+  static IndirectFitDataTest *createSuite() {
+    return new IndirectFitDataTest();
+  }
+
+  static void destroySuite(IndirectFitDataTest *suite) { delete suite; }
+
+  void test_data_is_instantiated_correctly() {
+    auto const workspace = WorkspaceCreationHelper::create2DWorkspace123(1, 3);
+    Spectra const spec =
+        std::make_pair(0u, workspace->getNumberHistograms() - 1);
+
+    workspace->setTitle("Test Title");
+    IndirectFitData const data(workspace, spec);
+
+    TS_ASSERT_EQUALS(data.workspace(), workspace);
+    TS_ASSERT_EQUALS(data.workspace()->getTitle(), "Test Title");
+    TS_ASSERT_EQUALS(data.workspace()->getNumberHistograms(), 1);
+  }
+
+  void test_that_DiscontinuousSpectra_is_set_correctly() {
+    DiscontinuousSpectra<std::size_t> const spectra =
+        DiscontinuousSpectra<std::size_t>("0-5,8,10");
+
+    std::string const spectraString = "0-5,8,10";
+    std::vector<std::size_t> const spectraVec{0, 1, 2, 3, 4, 5, 8, 10};
+
+    TS_ASSERT_EQUALS(spectra.getString(), spectraString);
+    for (auto it = spectra.begin(); it < spectra.end(); ++it)
+      TS_ASSERT_EQUALS(*it, spectraVec[it - spectra.begin()]);
+  }
+
+  void test_data_is_stored_correctly_in_the_ADS() {
+    auto const data = getIndirectFitData(1, 3);
+    SetUpADSWithWorkspace ads("WorkspaceName", data);
+
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("WorkspaceName"));
+    MatrixWorkspace_sptr workspace =
+        boost::dynamic_pointer_cast<MatrixWorkspace>(
+            AnalysisDataService::Instance().retrieve("WorkspaceName"));
+    TS_ASSERT_EQUALS(workspace->getNumberHistograms(), 1);
+  }
+
+  void test_displayName_returns_correct_name() {
+    auto const data = getIndirectFitData(1, 3);
+
+    std::vector<std::string> const formatStrings{
+        "%1%_s%2%_Result", "%1%_f%2%,s%2%_Parameter", "%1%_s%2%_Parameter"};
+    std::string const rangeDelimiter = "_to_";
+    std::size_t const spectrum = 1;
+
+    TS_ASSERT_EQUALS(data.displayName(formatStrings[0], rangeDelimiter),
+                     "_s0_Result");
+    TS_ASSERT_EQUALS(data.displayName(formatStrings[1], rangeDelimiter),
+                     "_f0+s0_Parameter");
+    TS_ASSERT_EQUALS(data.displayName(formatStrings[2], spectrum),
+                     "_s1_Parameter");
+  }
+
+  void test_displayName_removes_correct_part_of_workspace_name() {
+    auto const data = getIndirectFitData(1, 3);
+
+    SetUpADSWithWorkspace ads("Workspace_3456_red", data);
+    std::string const formatString = "%1%_s%2%_Result";
+    std::string const rangeDelimiter = "_to_";
+
+    TS_ASSERT_EQUALS(data.displayName(formatString, rangeDelimiter),
+                     "Workspace_3456_s0_Result");
+  }
+
+  void test_that_correct_number_of_spectra_is_returned() {
```

Comment:
Try to avoid using the work "correct" where possible. In this case a more descriptive name might be

`test_that_numberOfSpectra_returns_the_same_number_of_spectra_used_in_instantiation`